blocks/bricks/base.py

```diff
@@ -646,7 +647,8 @@ def initialize(self):
         if not self.initialization_config_pushed:
             self.push_initialization_config()
         for child in self.children:
-            child.initialize()
+            if not child.initialized:
+                child.initialize()
```

Comment:
On the other hand, maybe we should forbid both reallocation and reinitialization to keep things a bit simpler... needs more thinking.